contracts/MemeFactory.sol

```diff
@@ -333,11 +323,14 @@ abstract contract MemeFactory {
         // Calculate LP token allocation according to LP percentage and distribution to supporters
         Meme memeTokenInstance = Meme(memeToken);
         uint256 totalSupply = memeTokenInstance.totalSupply();
-        uint256 amountForLP = (totalSupply * LP_PERCENTAGE) / 100;
-        uint256 heartersAmount = totalSupply - amountForLP;
+        uint256 memeAmountForLP = (totalSupply * LP_PERCENTAGE) / 100;
+        uint256 heartersAmount = totalSupply - memeAmountForLP;
+
+        // Wrap native token to its ERC-20 version, where applicable
+        _wrap(nativeAmountForLP);
 
         // Create Uniswap pair with LP allocation
-        (address pool, uint256 liquidity) = _createUniswapPair(memeToken, totalNativeTokenCommitted, amountForLP);
+        (address pool, uint256 liquidity) = _createUniswapPair(memeToken, nativeAmountForLP, memeAmountForLP);
 
         // Record the actual meme unleash time
         memeSummon.unleashTime = block.timestamp;
```

Comment:
Below, in the collect, we used `totalNativeTokenCommitted`, however, this was no longer the correct value due to the adjustment above. I fixed this now by clearly keeping them separate.